metal/loadbalancers/emlb/emlb.go

```diff
@@ -10,24 +10,62 @@ import (
 )
 
 type LB struct {
-	client *lbaas.APIClient
+	client     *lbaas.APIClient
+	controller *controller
 }
 
 func NewLB(k8sclient kubernetes.Interface, config string) *LB {
+	metalAPIKey := "TODO"
 	lb := &LB{}
-	emlbConfig := lbaas.NewConfiguration()
-	lb.client = lbaas.NewAPIClient(emlbConfig)
+	lb.controller = NewController(metalAPIKey)
 	return lb
 }
 
 func (l *LB) AddService(ctx context.Context, svcNamespace, svcName, ip string, nodes []loadbalancers.Node) error {
+	// 1. Gather the properties we need: Metal API key, port number(s), cluster name(?), target IP(s?)
```

Comment:
For each function, I've scaffolded out what I think are the high-level steps so that we can work on filling things in in parallel.  For the `additionalProperties` and ` outputProperties` maps, the end goal is for them to be replaced with more specifically-named variables, but they are maps for now so we have flexibility to add/remove properties without too much impact on other code.